test/unit/test_conf.ml

```diff
@@ -0,0 +1,41 @@
+let test_check_version =
+  let test name ~expected ~exe res =
+    let test_name = "compare_version: " ^ name in
+    ( test_name
+    , `Quick
+    , fun () ->
+        let got = Ocamlformat_lib.Conf.check_version ~expected ~exe in
+        Alcotest.check Alcotest.(result unit string) test_name res got )
+  in
+  [ test "empty 'expected'" ~expected:"" ~exe:""
+      (Error "malformed version number \"\".")
+  ; test "invalid 'expected'" ~expected:"0.33..foo" ~exe:""
+      (Error "malformed version number \"0.33..foo\".")
+  ; test "invalid 'exe'" ~expected:"0.14.2" ~exe:"0.33..foo"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \
+          \"0.33..foo\".")
+  ; test "unknown 'exe'" ~expected:"0.14.2" ~exe:"unknown"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \"unknown\".")
+  ; test "equal" ~expected:"0.14.1" ~exe:"0.14.1" (Ok ())
+  ; test "accepted" ~expected:"0.14.1" ~exe:"0.14.1-15-g273f6f6-dirty"
```

Comment:
It makes sense to accept that but I think that the new semantics should be documented somewhere. Maybe a new section in the manpage (since only CLI options are documented, the fact that you can put `version=` in a config file is actually not directly documented)